searchEngine/SearchEngine.py

```diff
@@ -0,0 +1,55 @@
+from MockArticle import Article
+
+class SearchEngine:
+    """
+    This class implements the functions to score every article 
+    by the amount of incidences between keywords and synonyms to articles
+
+    Attributes:
+        keywords: A list of keywords
+        synonyms: A list of synonyms for the keywords
+        articles: A list of articles
+        keywords_weight: The weight that a keyword has on the scorig algorithm
+            Optional: default argument as 1
+        synonyms_weight: The weight that a synonym has on the scoring algorithm
+            Optional: default argument as 1
+    """
+    def __init__(self, articles):
```

Comment:
Something to consider is if you want `keywords_weight` and `synonyms_weight` to be a configuration of the entire `SearchEngine`, or a per-query configuration.

In general, allowing more fine-grained configuration is nice because it's more flexible, but too much flexibility can also cause end users to become overwhelmed or confused. So you just want to find the balance that you think works best.